test/TestDestroy.cpp

```diff
@@ -27,17 +27,27 @@ TEST(TestDestroy, TestDestroyTensorSingle)
         {
             kp::Manager mgr;
 
-            tensorA = mgr.tensor({ 0, 0, 0 });
+            const std::vector<float> initialValues = {0.0f, 0.0f, 0.0f};
+
+            tensorA = mgr.tensor(initialValues);
 
             std::shared_ptr<kp::Algorithm> algo =
-              mgr.algorithm({ tensorA }, spirv);
+              mgr.algorithm({tensorA}, spirv);
+
+            // Sync values to and from device
+            mgr.sequence()
+            ->eval<kp::OpTensorSyncDevice>(algo->getTensors())
+            ->eval<kp::OpTensorSyncLocal>(algo->getTensors());
+
+            EXPECT_EQ(tensorA->vector(), initialValues);
 
             mgr.sequence()
               ->record<kp::OpAlgoDispatch>(algo)
               ->eval()
               ->eval<kp::OpTensorSyncLocal>(algo->getTensors());
```

Comment:
Here is where you can see the values are copied back from the GPU vRAM

Reply:
Thanks for walking me through it.